awsmp/models.py

```diff
@@ -139,3 +142,157 @@ def ami_access_role_arn_validator(cls, value):
+    SupportInformation: SupportInformationModel
+    RegionAvailability: RegionAvailabilityModel
+    SupportTerm: str
+
+    @staticmethod
+    def get_entity(response: dict[str, Any]):
+        return EntityModel(**response)
+
+    @staticmethod
+    def get_entity_from_yaml(yaml_config: dict[str, Any]):
+        desc = AmiProduct(**yaml_config["description"])
+        region = Region(**yaml_config["region"])
+        refund_policy = yaml_config["refund_policy"]
+
+        yaml_to_api_response = {
+            "Description": {
+                "ProductTitle": desc.product_title,
+                "ShortDescription": desc.short_description,
+                "LongDescription": desc.long_description,
+                "Sku": desc.sku,
+                "Highlights": desc.highlights,
+                "SearchKeywords": desc.search_keywords,
+                "Categories": desc.categories,
+            },
+            "PromotionalResources": {
+                "LogoUrl": desc.logourl,
+                "Videos": desc.video_urls,
+                "AdditionalResources": desc.additional_resources,
+            },
+            "SupportInformation": {
+                "Description": desc.support_description,
+                "Resources": desc.support_resources,
+            },
+            "RegionAvailability": {
+                "Regions": region.commercial_regions,
+                "FutureRegionSupport": region.future_region_support[-1],
+            },
+            "SupportTerm": refund_policy,
+        }
+
+        return EntityModel(**yaml_to_api_response)
+
+    @staticmethod
+    def get_diff_model_type(field_name: str, live_listing_value: Any, local_config_value: Any):
+        # fields which doesn't send API request from ordering diference
+        non_ordered_fields = ["Regions"]
+
+        if not live_listing_value and local_config_value:
+            return DiffAddedModel(field_name=field_name, added_value=local_config_value)
+        elif live_listing_value and not local_config_value:
+            return DiffRemovedModel(field_name=field_name, removed_value=live_listing_value)
+        else:
+            if field_name in non_ordered_fields:
+                differ_value = set(local_config_value) != set(live_listing_value)
+            else:
+                differ_value = live_listing_value != local_config_value
+
+            if differ_value:
+                return DiffChangedModel(
+                    field_name=field_name, old_value=live_listing_value, new_value=local_config_value
+                )
+
+    @staticmethod
+    def add_to_diff_list(res, diff_added, diff_removed, diff_changed):
+        if isinstance(res, DiffAddedModel):
+            diff_added.append(res)
+        elif isinstance(res, DiffRemovedModel):
+            diff_removed.append(res)
+        elif isinstance(res, DiffChangedModel):
+            diff_changed.append(res)
+
+        return diff_added, diff_removed, diff_changed
```

Comment:
since these objects are mutable (and mutated) we don't actually need to return anything here.